### PR TITLE
add send-analytics flag

### PR DESCRIPTION
**Pull request by elharo**
@suztomo in prep for #1619

### CODE REVIEW

File: linkage-monitor/pom.xml [hunk 1]
```diff
@@ -78,6 +78,11 @@
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
     </dependency>
+	<dependency>
+	  <groupId>commons-cli</groupId>
+	  <artifactId>commons-cli</artifactId>
+	  <version>1.4</version>
```
Comment (suztomo) [inline linkage-monitor/pom.xml:84]: Suggestion: how about adding commons-cli to the root pom's dependency management section? The "dependencies" module also uses it.

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 2]
```diff
@@ -85,17 +90,22 @@
   private final ImmutableMap<String, String> localArtifacts =
       findLocalArtifacts(repositorySystem, session, Paths.get(".").toAbsolutePath());
 
-  public static void main(String[] arguments)
+  public static void main(String[] args)
```
Comment (suztomo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java:93]: Google rule is not to abbreviate words.

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 3]
```diff
@@ -111,6 +121,29 @@ public static void main(String[] arguments)
     }
   }
 
+  private static CommandLine parseCommandLine(String[] arguments) {
+    Options options = new Options();
+    options.addOption("send-analytics", false, "track results from each run");
```
Comment (suztomo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java:126]: Would you make this more explicit about this tool uploading the result to Google Analytics.

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 4]
```diff
@@ -111,6 +121,29 @@ public static void main(String[] arguments)
...
+    } catch (ParseException ex) {
+      printUsageAndDie();
+      return null;
+    }
+  }
+
+  private static void printUsageAndDie() {
+    logger.severe(
+        "Usage: java com.google.cloud.tools.dependencies.linkagemonitor.LinkageMonitor"
+        + " --[no]send-analytics com.google.cloud:libraries-bom");
```
Comment (suztomo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java:143]: I don't see option for `--nosend-analytics` (yet?)

**Review by suztomo — APPROVED** [review]

File: linkage-monitor/pom.xml (hunk 1 above)
Comment (elharo) [reply linkage-monitor/pom.xml]: good idea, done

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java (hunk 2 above)
Comment (elharo) [reply linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: done

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java (hunk 3 above)
Comment (elharo) [reply linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java:124]: what language do you suggest? This needs to be a short string for help, not complete documentation.

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java (hunk 4 above)
Comment (elharo) [reply linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: I'm too used to gnu. Fixed.

**Pull request merged**